index.js

```diff
@@ -23,14 +23,57 @@ exports.serialize = serialize;
 var __toString = Object.prototype.toString
 
 /**
- * RegExp to match field-content in RFC 7230 sec 3.2
+ * RegExp to match cookie-name in RFC 6265 sec 4.1.1
+ * This refers out to the obsoleted definition of token in RFC 2616 sec 2.2
+ * which has been replaced by the token definition in RFC 7230 appendix B.
  *
- * field-content = field-vchar [ 1*( SP / HTAB ) field-vchar ]
- * field-vchar   = VCHAR / obs-text
- * obs-text      = %x80-FF
+ * cookie-name       = token
+ * token             = 1*tchar
+ * tchar             = "!" / "#" / "$" / "%" / "&" / "'" /
+ *                     "*" / "+" / "-" / "." / "^" / "_" /
+ *                     "`" / "|" / "~" / DIGIT / ALPHA
  */
 
-var fieldContentRegExp = /^[\u0009\u0020-\u007e\u0080-\u00ff]+$/;
+var cookieNameRegExp = /^[!#$%&'*+.^_`|~0-9A-Za-z-]+$/;
```

Comment:
This is very much a nitpick, sorry, but could we maintain the ordering from the spec section in the regex and use escaping instead of the "end" trick? I just find that trick always throws me off when trying to visually parse groups like this which also include ranges.

```suggestion
var cookieNameRegExp = /^[!#$%&'*+.^_\-`|~0-9A-Za-z]+$/;
```

Reply:
I've no objections to fixing it that way, makes sense; but NB I didn't commit your suggestion because that's not the order in the spec either!